### ISSUE TITLE
WiFi.localIP() returns 0.0.0.0 after saving config

### ISSUE THREAD

**Issue by martinberlin** [martinberlin/FS2#11]
https://github.com/martinberlin/FS2/blob/0d4eb1a062c7352d57f31dbb37a72306553cc579/FasarekFS2/FasarekFS2.ino#L452

Hello @tablatronix 
I have a question regarding this. It happens only after setting a new WiFi, it saves config, and on restart WiFi.localIP() returns 0.0.0.0 but it has a correct IP assigned. 
Is this just question of making a delay(100) or I'm missing something ?

Thanks!

**Comment by tablatronix** [comment]
on esp32 local ip does not exist until wifi is started, I think I added a way to get this in wm, I will see if I can expose it. try the wm getlocalip function if you can find it

**Comment by tablatronix** [comment]
I think I was mistaken, I was thinking about ssid.

esp32 does not start wifi automatically like esp8266 does

**Comment by martinberlin** [comment]
Ahem, no problem. But can you reproduce this ? 
I will make an ESP32 version in [the new repository](https://github.com/martinberlin/FS32) much more clean without so many "options" that at the end turn to be buggy. Like you told me just autoconnect and maybe offer a forced restart mode if you want to update the WiFi but all settings are going to be moved out.

**Comment by tablatronix** [comment]
Are you calling autoconnect or begin? Because wifi will not connect on esp32

**Comment by martinberlin** [comment]
I'm calling autoconnect.
https://github.com/martinberlin/FS32/blob/master/src/FasarekFS2.ino#L309

But is strange since it autoconnects on the second try. It does just deliver:
0.0.0.0  only after saveParamsCallback is triggered. Otherwise connects correctly and the IP is rendered on the display.

**Comment by tablatronix** [comment]
ok can you post a reproduce list, which codebase and what steps, sounds like a bug

**Comment by martinberlin** [comment]
Codebase is here:
https://github.com/martinberlin/FS32
Is the same code as the other just in this Repository I'm using PlatformIO as editor and trying to leave only the essential. I'm also removing all camera initialization from setup() just like you mentioned before. 

With this code is just uploading SPIFFS and then the code to an ESP32. Dependencies [are here](https://github.com/martinberlin/FS32/blob/master/platformio.ini)).

**Comment by martinberlin** [comment]
No, just master at this time equal to develop. I will code further in develop.

**Comment by tablatronix** [comment]
You have to use straight camera modules no spi if you want something better. Which is possible on esp32 because it has enough pins

**Comment by martinberlin** [comment]
Sounds interesting, can you send me a link to one of this straight camera modules please ? Thanks Shawn

**Comment by tablatronix** [comment]
They are not modules they are bare cameras

**Comment by martinberlin** [comment]
And how it's supposed to be wired to the ESP32 ? It's possible to read an image somehow like this ?
I will research further. It's very strange the only one producing cameras for hobbysts/makers is just one single company in China.

**Comment by tablatronix** [comment]
The esp32 needs to have psram so it can buffer enough memory, 5-8mp is alot of data for even an esp look at some of the raspi cameras same thing
